### PR TITLE
feat(Project): Added Vulnerability Summary Tab in Projects.

### PR DESCRIPTION
**Pull request by rudra-superrr**
Signed-off-by: rudra-superrr <rudra.chopra@siemens.com>

Description: Vulnerability Summary Tab in Projects will includes vulnerabilities of all the projects (parent+subProjects).

Issue: 

### How To Test?
- Goto Project Portlet
- Then select Vulnerability tab
- There you will see Summary tab which contains total vulnerabilities

Note: If the parent project does not have any subProjects then this Summary tab will not appear.

image

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.

image
image
image

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 1]
```diff
@@ -1603,6 +1603,17 @@ private void prepareDetailView(RenderRequest request, RenderResponse response) t
                 Collection<ProjectLink> links = allSubProjectLinks.stream().collect(
                         Collectors.toMap(ProjectLink::getId, Function.identity(), (oldValue, newValue) -> oldValue)).values();
                 request.setAttribute(ALL_SUB_PROJECT_LINK, links);
+                List<String> ProjIds = links.stream().map(proj->proj.id).collect(Collectors.toList());
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:1606]: Please check the variable case.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 2]
```diff
@@ -1603,6 +1603,17 @@ private void prepareDetailView(RenderRequest request, RenderResponse response) t
...
+                VulnerabilityService.Iface vulClient = thriftClients.makeVulnerabilityClient();
+                List<VulnerabilityDTO> total_vuls = new ArrayList<VulnerabilityDTO>();
+                for (int pos=0; pos<ProjIds.size(); pos++) {
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:1609]: indentation

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            $('#SummaryPage').show()
+        });
+        
+        function createVulnerabilitySummaryTable() {
+            var table,
+                tableDefinition,
+
+            tableDefinition = {
+                columns: [
+                	{ title: "<liferay-ui:message key="project.name" />"},
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp:89]: indentation

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp [hunk 4]
```diff
@@ -0,0 +1,110 @@
+<%--
+  ~ Copyright Siemens AG, 2016-2017, 2019, 2021, 2023. Part of the SW360 Portal Project.
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp:2]: Please change the copyright

**Review by smrutis1 — CHANGES_REQUESTED** [review]

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java (hunk 1 above)
Comment (rudra-superrr) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java]: made the changes

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java (hunk 2 above)
Comment (rudra-superrr) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java]: done

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp (hunk 3 above)
Comment (rudra-superrr) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp]: done

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp (hunk 4 above)
Comment (rudra-superrr) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/totalVulnerabilities.jsp]: done

**Review by smrutis1 — APPROVED** [review]
Code looks good

**Pull request merged**